src/modules/perl.rs

```diff
@@ -0,0 +1,102 @@
+use super::{Context, Module, RootModuleConfig, SegmentConfig};
+
+use crate::configs::perl::PerlConfig;
+use crate::utils;
+
+/// Creates a module with the current perl version
+///
+/// Will display the perl version if any of the following criteria are met:
+///     - Current directory contains a `.pl` or a `.pm` file
+///     - Current directory contains a "Makefile.PL", "cpanfile", "META.json", "META.yml", or ".perl-version" file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_perl_project = context
+        .try_begin_scan()?
+        .set_files(&[
+            "Makefile.PL",
+            "cpanfile",
+            "META.json",
+            "META.yml",
+            ".perl-version",
+        ])
+        .set_extensions(&["pl", "pm"])
+        .is_match();
+
+    if !is_perl_project {
+        return None;
+    }
+
+    let perl_version = utils::exec_cmd("perl", &["-e", "print $^V;"])?.stdout;
```

Comment:
Your command does not match with your mock command. So change either one to match.